src/main/java/com/acme/dbo/txlog/WriterFactory.java

```diff
@@ -0,0 +1,8 @@
+package com.acme.dbo.txlog;
+
+public class WriterFactory {
```

Comment:
It's not Factory )) Factory _returns_ writer. Instead your factory do _write_ so have to be Writer

Reply:
Isn't it a Factory pattern? Why not naming it as Factory as we seemed to be doing so in our cases? We used to add pattern to our name.

Reply:
Nope ) We name something as Factory only if _gives_ you some other objects. Your class doesn't give any objects, it just writes ) So it's Writer not WriterFactory.

Reply:
Resolved.